langchain4j-pgvector/src/main/java/dev/langchain4j/store/embedding/pgvector/MetadataConfig.java

```diff
@@ -0,0 +1,50 @@
+package dev.langchain4j.store.embedding.pgvector;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Metadata configuration.
+ */
+public interface MetadataConfig {
+    /**
+     * Metadata type
+     * <ul>
+     * <li>COLUMNS: for static metadata, when you know in advance the list of metadata
+     * <li>JSON: For dynamic metadata, when you don't know the list of metadata that will be used.
+     * <li>JSONB: Same as JSON, but stored in a binary way. Optimized for query on large dataset.
+     * </ul>
+     * <p>
+     * Default value: JSON
+     * @return Metadata type
+     */
+    String type();
+    /**
+     * SQL definition of metadata field(s).
+     * By default, "metadata JSON NULL" configured for JSON metadata type
+     * Ex: condominium_id uuid null, user uuid null
+     * @return Metadata Definition
+     */
+    List<String> definition();
```

Comment:
```suggestion
    Map<String, Class<?>> columnDefinitions();
```
I am wondering: if `SQL_TYPE_MAP` is used to map types dynamically, can we use the same approach here?
Then it could be simplified to:
```java
.columnDefinitions(Map.of(
    "name", String.class,
    "age", Integer.class,
    ...
))
```

Or, maybe even better:
```java
    Map<String, ColumnDefinition> columnDefinitions();
```
where `ColumnDefinition` will contain column name (can be different from metadata key name) and column type